honeybee_grasshopper_energy/src/HB ProgramType.py

```diff
@@ -0,0 +1,88 @@
+            starting point for the new ProgramType output from this component.
+            This can also be text for the name of a ProgramType within the library
+            such as that output from the "HB Search Program Types" component.
+            If None, a Plenum program type will be used as the base with no loads,
+            setpoints, or ventilation requirements assigned.
+        _people_: A People object to describe the occupancy of the program. If None,
+            no occupancy will be assumed for the program. Default: None.
+        _lighting_: A Lighting object to describe the lighting usage of the program.
+            If None, no lighting will be assumed for the program. Default: None.
+        _electric_equip_: An ElectricEquipment object to describe the usage
+            of electric equipment within the program. If None, no electric equipment
+            will be assumed for the program. Default: None.
+        _gas_equip_: A GasEquipment object to describe the usage of gas equipment
+            within the program. If None, no gas equipment will be assumed for
+            the program. Default: None.
+        _infiltration_: An Infiltration object to describe the outdoor air leakage of
+            the program. If None, no infiltration will be assumed for the program.
+            Default: None.
+        _ventilation_: A Ventilation object to describe the minimum outdoor air
+            requirement of the program. If None, no ventilation requirement will
+            be assumed for the program. Default: None
+        _setpoint_: A Setpoint object to describe the temperature and humidity
+            setpoints of the program.  If None, the ProgramType cannot be assigned
+            to a Room that is conditioned. Default: None.
+    
+    Returns:
+        program: A ProgramType object that can be assigned to Honeybee Rooms in
+            order to specify all default schedules and loads on the Room.
+"""
+
+ghenv.Component.Name = "HB ProgramType"
+ghenv.Component.NickName = 'ProgramType'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Energy"
+ghenv.Component.SubCategory = '0 :: Basic Properties'
+ghenv.Component.AdditionalHelpFromDocStrings = "2"
+
+try:
+    from honeybee_energy.programtype import ProgramType
+    from honeybee_energy.lib.programtypes import program_type_by_name
+    from ladybug_rhino.grasshopper import all_required_inputs
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee:\n\t{}'.format(e))
+
+
+if all_required_inputs(ghenv.Component):
+    # get the base program type
+    if base_program_ is None:
+        program = ProgramType(_name)
+    else:
+        if isinstance(base_program_, str):
+            base_program_ = program_type_by_name(base_program_)
+        program = base_program_.duplicate()
+        program.name = _name
+    
+    # go through each input load and assign it to the set
+    if _people_ is not None:
+        program.people = _people_
```

Comment:
Yes, the program assignment does handle `None` values.  The reason why we have these `if` statements here is because we allow people to start from an input `base_program_` if they don't want to make a new `ProgramType` object from scratch. So, if we didn't do this check, inputting `None` with the program assignment would overwrite whatever was on the `base_program_` to no longer exist.

Reply:
BTW, if a user really wants one of the loads to be `None`, they can achieve this by not inputting a `base_program_`.